src/main/java/demo/multiagent/application/AgenticWorkflow.java

```diff
@@ -175,12 +167,10 @@ private Step planExecution() {
   private Step runPlan() {
     return step(EXECUTE_PLAN)
       .call(() -> {
-
         var stepPlan = currentState().nextStepPlan();
         logger.debug("Executing plan step (agent:{}), asking {}", stepPlan.agentId(), stepPlan.query());
-        var agent = agentsRegistry.getAgent(stepPlan.agentId());
-
-        var agentResponse = agent.query(commandContext().workflowId(), stepPlan.query());
+        var agentCall = agentCall(stepPlan.agentId());
+        var agentResponse = agentCall.invoke(stepPlan.query());
```

Comment:
I think I would prefer to have a single method: `var response = callAgent(agentId, query)`